Sources/Apodini/Semantic Model Builder/GraphQL/GraphQLSchemaBuilder.swift

```diff
@@ -0,0 +1,248 @@
+            let response: EventLoopFuture<Response<AnyEncodable>> = mutableContext.handle(request: request, eventLoop: vaporRequest.eventLoop.next())
+
+            return response.flatMapThrowing { encodableAction -> Any? in
+                switch encodableAction {
+                case let .send(element),
+                     let .final(element):
+                    return element.wrappedValue
+                case .nothing, .end:
+                    return ".nothing, .end"
+                }
+            }
+
+        })
+    }
+
+
+    // Generated adjacency list tree
+    func append<H: Handler>(for endpoint: Endpoint<H>, with context: AnyConnectionContext<GraphQLInterfaceExporter>) {
+        // Remove parameters from the path by using `":" filter`
+        var currentPath = endpoint.absolutePath.map {
+            $0.description.lowercased()
+        }.filter {
+            $0.first != ":"
+        }
+
+        // Remove `root`
+        currentPath.removeFirst()
+        // Remove `v1`
+        currentPath.removeFirst()
+
+        print("The current path is ->", currentPath)
+
+        // Create node names
+        var currentSum = String()
+        if (currentPath.count > 1) {
+            for ix in 0..<currentPath.count {
+                currentSum.append(currentPath[ix])
+                currentSum.append("_")
+                currentPath[ix] = currentSum
+            }
+        }
+
+        // Get leaf name
+        let leafName = currentPath.last ?? "None"
+
+        // Handle arguments
+        for p in endpoint.parameters {
+            let graphqlType = graphqlTypeMap(with: p.propertyType)
+            if (p.necessity == .required) {
+                self.args[leafName, default: [:]][p.name] = GraphQLArgument(type: GraphQLNonNull(graphqlType), description: p.description)
+            } else {
+                self.args[leafName, default: [:]][p.name] = GraphQLArgument(type: graphqlType, description: p.description)
+            }
+
+        }
+
+        // Response type and context info
+        let treeTemp = try! EnrichedInfo.node(endpoint.handleReturnType)
+        self.responseTypeTree[leafName] = treeTemp
+        self.leafContext[leafName] = context
+
+        // Handle Single points
+        if (currentPath.count == 1) {
+            self.fields[leafName] = self.graphQLFieldCreator(for: self.responseTypeTree[leafName]!,
+                    self.leafContext[leafName]!,
+                    self.args[leafName] ?? [:])
+            return
+        }
+
+        // Create tree
+        var indx = currentPath.count - 1
+        while (indx >= 1) {
+            let child = currentPath[indx], parent = currentPath[indx - 1]
+            if (self.tree.keys.contains(parent)) {
+                self.tree[parent]!.insert(child)
+            } else {
+                self.tree[parent] = [child]
+            }
+            hasIncomingEdge.insert(child)
+            indx -= 1
+        }
```

Comment:
To be honest I am a bit lost about the functionality at the end of the function here. It would be great to split the function out into smaller functions or types. Maybe some comments would also help a bit.